orchestration/templates/ingest-table.yaml

```diff
@@ -0,0 +1,120 @@
+apiVersion: argoproj.io/v1alpha1
+kind: WorkflowTemplate
+metadata:
+  name: ingest-table
+spec:
+  templates:
+    - name: main
+      inputs:
+        parameters:
+          - name: url
+          - name: dataset-id
+          - name: gcs-bucket
+          - name: gcs-prefix
+          - name: table-name
+          - name: timeout
+          - name: sa-secret
+          - name: sa-secret-key
+      dag:
+        tasks:
+          # 1) Make a post request to the Jade API to ingest the table
+          - name: submit-table-ingest
+            template: submit-table-ingest
+            arguments:
+              parameters:
+                - name: url
+                  value: '{{ "{{inputs.parameters.url}}" }}'
+                - name: dataset-id
+                  value: '{{ "{{inputs.parameters.dataset-id}}" }}'
+                - name: source-path
+                  value: '{{ "gs://{{inputs.parameters.gcs-bucket}}/{{inputs.parameters.gcs-prefix}}" }}'
+                - name: table-name
+                  value: '{{ "{{inputs.parameters.table-name}}" }}'
+                - name: sa-secret
+                  value: '{{ "{{inputs.parameters.sa-secret}}" }}'
+                - name: sa-secret-key
+                  value: '{{ "{{inputs.parameters.sa-secret-key}}" }}'
+          # 2) Poll the ingest job for the table repeatedly until the table has been ingested
+          - name: poll-ingest-job
+            dependencies: [submit-table-ingest]
+            template: poll-ingest-job
+            arguments:
+              parameters:
+                - name: url
+                  value: '{{ "{{inputs.parameters.url}}" }}'
+                - name: job-id
+                  value: '{{ "{{tasks.submit-table-ingest.outputs.result}}" }}'
+                - name: timeout
+                  value: '{{ "{{inputs.parameters.timeout}}" }}'
+                - name: sa-secret
+                  value: '{{ "{{inputs.parameters.sa-secret}}" }}'
+                - name: sa-secret-key
+                  value: '{{ "{{inputs.parameters.sa-secret-key}}" }}'
+    {{- $secretVolume := "sa-secret-volume" }}
+    {{- $mountPath := "/secret" }}
+    # SubTemplates
+    - name: submit-table-ingest
+      inputs:
+        parameters:
+          - name: url
+          - name: dataset-id
+          - name: source-path
+          - name: table-name
+          - name: sa-secret
+          - name: sa-secret-key
+      volumes:
+        - name: '{{ $secretVolume }}'
+          secret:
+            secretName: '{{ .Values.repo.accessKey.secretName }}'
+      script:
+        image: us.gcr.io/broad-dsp-gcr-public/monster-auth-req-py:1.0.0
```

Comment:
Should these be `1.0.0`? I thought we had a newer version

Reply:
YES Good catch